docs/designs/checkpoint_design.md

```diff
@@ -0,0 +1,59 @@
+# ElasticDL Training Checkpoints Design
+This document describes the design of save checkpoints for ElasticDL training.
+
+## Motivation
+Checkpoints capture the exact value of all parameters (tf.Variable objects and embedding vectors) used by a model. In ElasticDL, we need to save all parameters of the model to checkpoints during training for the following reasons:
```

Comment:
Checkpoint stores the state of a training process, it includes not only variables but also certain states like `global_step`.